### ISSUE TITLE
jdl blueprint errors when generating sample application using dev blueprint

### ISSUE THREAD

**Issue by dwarakaprasad** [jhipster/generator-jhipster#25489]
##### **Overview of the issue**
A jdl blueprint errors when run using the .blueprint/generators/generate-sample/generator.mjs
command: `jhipster-<blueprintName> generate-sample sample.jdl`

```
No change to package.json was detected. No package manager install will be executed.
ERROR! An error occured while running jhipster-adikkal:jdl#checkOptions
ERROR! ERROR! At least one jdl file is required.
Error: At least one jdl file is required.
    at default.checkOptions (file:///home/dwarka/develop/code/brmaStr/basepack/generator-jhipster-adikkal/node_modules/generator-jhipster/dist/generators/jdl/generator.js:88:27)
    at default.executeTask (file:///home/dwarka/develop/code/brmaStr/basepack/generator-jhipster-adikkal/node_modules/yeoman-generator/dist/actions/lifecycle.js:244:26)
    at env.queueTask.once (file:///home/dwarka/develop/code/brmaStr/basepack/generator-jhipster-adikkal/node_modules/yeoman-generator/dist/actions/lifecycle.js:218:56)
    at runLoop.add.once (file:///home/dwarka/develop/code/brmaStr/basepack/generator-jhipster-adikkal/node_modules/yeoman-environment/dist/environment-base.js:395:23)
    at Immediate.<anonymous> (/home/dwarka/develop/code/brmaStr/basepack/generator-jhipster-adikkal/node_modules/grouped-queue/lib/subqueue.js:48:34)
    at process.processImmediate (node:internal/timers:478:21)
```

##### **Motivation for or Use Case**
It should be possible to use generate-sample devBlueprint to generate samples.

##### **Reproduce the error**
Create a blueprint for jdl and try running the command
`jhipster-<blueprintName> generate-sample sample.jdl`

##### **Related issues**
None.

##### **Suggest a Fix**

Change [this](https://github.com/jhipster/generator-jhipster/blob/244c7fedcda6481adf16b87193d8ff6bb8ed1756/generators/jdl/generator.ts#L79) to 

`await this.composeWithBlueprints('jdl', { generatorArgs: (this.options as any).jdlFiles });`

similar to language [generator](https://github.com/jhipster/generator-jhipster/blob/244c7fedcda6481adf16b87193d8ff6bb8ed1756/generators/languages/generator.js#L71)

jdl generator can be composed as follows,

```
await this.composeWithJHipster('jdl', {
          generatorArgs: [this.sampleName],
          generatorOptions: {
            skipJhipsterDependencies: true,
            skipInstall: true,
            jdlFiles: [ this.sampleName ],
          },
        });
```

##### **JHipster Version(s)**
main branch

##### **JHipster configuration**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "additionalSubGenerators": "",
    "baseName": "testblue",
    "cli": true,
    "entities": [],
    "generators": {
      "jdl": {
        "command": false,
        "priorities": [
          "initializing",
          "prompting",
          "configuring",
          "composing",
          "loading",
          "preparing",
          "configuringEachEntity",
          "loadingEntities",
          "preparingEachEntity",
          "preparingEachEntityField",
          "preparingEachEntityRelationship",
          "postPreparingEachEntity",
          "default",
          "writing",
          "writingEntities",
          "postWriting",
          "postWritingEntities",
          "loadingTranslations",
          "install",
          "postInstall",
          "end"
        ],
        "sbs": false,
        "written": true
      },
    },
    "jhipsterVersion": "8.1.0",
    "localBlueprint": false,
    "sampleWritten": true,
    "subGenerators": [
      "jdl"
    ]
  }
}
</pre>
</details>

##### **Environment and Tools**

openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Temurin-17.0.8+7 (build 17.0.8+7)
OpenJDK 64-Bit Server VM Temurin-17.0.8+7 (build 17.0.8+7, mixed mode, sharing)

git version 2.34.1

node: v20.11.1
npm: 10.2.4

Docker version 25.0.3, build 4debf41

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Browsers and Operating System**

- [X] Checking this box is mandatory (this is just to show you read everything)

**Comment by mshima** [comment]
The languages implementation was old when args and options was defined in constructor.
https://github.com/jhipster/generator-jhipster/pull/25497 should be a more generic solution.

**Comment by mshima** [comment]
@dwarakaprasad let me know if the solution is ok.

**Comment by dwarakaprasad** [comment]
> @dwarakaprasad let me know if the solution is ok.

@mshima this is much better than what I had suggested. I was a little hesitant to use _agrs (although it was used in parseJHipsterArguments) from yeoman. Your solution has fixed the problem for all blueprint generators. This is great and thank you so much.

I tested both cli & generate-sample, both works like a charm. Once again thank you so much...
